### ISSUE TITLE
Install deno with homebrew

### ISSUE THREAD

**Issue by ferried** [denoland/deno#1486]
My mac version is macOS Mojave.10.14.3

**Comment by hayd** [comment]
https://github.com/Homebrew/homebrew-core/pull/35645

**Comment by ferried** [comment]
> [Homebrew/homebrew-core#35645](https://github.com/Homebrew/homebrew-core/pull/35645)

thanks

**Comment by ry** [comment]
Reopening to track progress

**Comment by raphaelsoul** [comment]
does this build v0.2.6 still requires llvm when installing with brew?

> * Dependency 'llvm' may be unnecessary as it is provided by macOS; try to build this formula without it.

progress stucks for nearly 3 weeks

**Comment by hayd** [comment]
Yup. I will bump the version to 0.2.8 and see if anyone picks up the review...

It's _possible_ that it needs to be refactored to not use `python tools/setup.py` and `python tools/build.py` i.e. unroll it to explicit gn&ninja commands, but I'd rather avoid doing this and I have not been asked to do this. I think I have addressed the first reviewers concerns...

**Comment by hayd** [comment]
FYI you can use my basic brew formula to get 0.2.8:
```
brew install https://gist.githubusercontent.com/hayd/e6ea4568883d339b1787509cb74d0726/raw/4865d27d23745cb164a297096926e7da707f7fff/deno.rb
```

**Comment by ry** [comment]
almost there... https://github.com/Homebrew/homebrew-core/pull/35645

**Comment by hayd** [comment]
@ry it sounds like we may need to resolve https://github.com/denoland/deno/issues/1474 to get it past the finish line.

Any tips on this? Specifically if no prebuilts are used (with a flag to setup or build) what would need to change in gn?

**Comment by hayd** [comment]
Ah, seen you've already replied. Sorry for the noise!

**Comment by szhu** [comment]
I wonder if https://github.com/Homebrew/homebrew-cask will accept a binary version of a command-line program.

Also, even if they don't accept these formulae, you could always just create a tap. Installing it from a tap is still a one-liner: `brew install denoland/deno/deno` or `brew cask install denoland/deno/deno` depending on whether you want to ship the binary.

**Comment by hayd** [comment]
Please give it a try! Below is a binary brew formula, though you may want to replace the test from the homebrew-core PR. That said, I think this needs a little tweaking to be a valid cask...

https://gist.github.com/hayd/e6ea4568883d339b1787509cb74d0726

**Comment by hayd** [comment]
`brew install deno` now works!

```
 % brew install deno
==> Downloading https://homebrew.bintray.com/bottles/deno-0.6.0.mojave.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/47/470aa4ad260593875fff68af034a574d447ddeeacb0a0f6ca5b99052c196cec0?__gda__=exp=1558645609~hmac
######################################################################## 100.0%
==> Pouring deno-0.6.0.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/deno/0.6.0: 5 files, 48.7MB
 % deno version
deno: 0.6.0
v8: 7.6.53
typescript: 3.4.1
```

**Comment by ferried** [comment]
> Please give it a try! Below is a binary brew formula, though you may want to replace the test from the homebrew-core PR. That said, I think this needs a little tweaking to be a valid cask...
> 
> https://gist.github.com/hayd/e6ea4568883d339b1787509cb74d0726

that's mine
```
macross:~ ferried$ brew search deno
==> Formulae
denominator

==> Casks
code-notes                               code-notes
macross:~ ferried$ 
```
what should i do that i can searched it in brew,`tap?`

**Comment by hayd** [comment]
@ferried it's in mainline homebrew-core no need for cask/tap!

`brew update` might be required?

**Comment by bartlomieju** [comment]
Can be closed now?

CC @ry

**Comment by Gopaljeepurwar** [comment]
Getting error while installing deno using Homebrew on macOS

Error: No available formula with the name "deno"

**Comment by kt3k** [comment]
@Gopaljeepurwar Did you run `brew update`?

**Comment by ry** [comment]
https://formulae.brew.sh/formula/deno

**Comment by pierredewilde** [comment]
Upgrading deno with homebrew from 0.17 to 0.18 on Mac OS 10.14.6 requires 2 new dependencies:

```==> Upgrading 1 outdated package:
deno 0.17.0 -> 0.18.0
==> Upgrading deno 
==> Installing dependencies for deno: swig and llvm
```

**Comment by pierredewilde** [comment]
Fixed with 0.18.0_1. Thanks.

**Comment by tannakartikey** [comment]
I am getting this error on Ubuntu 18.

Below are a few details:
```
 ✘  ~  brew install deno
deno: macOS is required.
Error: An unsatisfied requirement failed this build.
 ✘  ~  brew --version
Homebrew 2.2.11
Homebrew/linuxbrew-core (git revision 81b88; last commit 2020-03-30)
 ~  lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
 ~  brew info deno
deno: stable 0.36.0
Command-line JavaScript / TypeScript engine
https://deno.land/
Not installed
From: https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/deno.rb
==> Dependencies
Build: llvm ✘, ninja ✘, rust ✘, xz ✘
Required: python@2 ✘, xz ✘
==> Requirements
Required: macOS is required ✘
==> Analytics
install: 8 (30 days), 140 (90 days), 214 (365 days)
install-on-request: 8 (30 days), 139 (90 days), 213 (365 days)
build-error: 0 (30 days)
```
